### PR TITLE
stage 2 done

### PR DESCRIPTION
**Pull request by accountF**

### CODE REVIEW

File: sources/data/itemDataActivity.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+class ItemDataActivity {
```
Comment (origamirin) [inline sources/data/itemDataActivity.js]: This class is identical to ItemDataContact. Create one class and export two different object from two different files

File: sources/views/activities/formActivity.js [hunk 2]
```diff
@@ -89,29 +94,38 @@ export default class formForAddActivity extends JetView {
...
+			const item = ItemDataActivity.getItem();
 			item.Time = item.DueDate;
 			this.formComponent.setValues(item);
 		}
+
+		if (this.url[0].page !== "activities") {
+			let idFromContact = +this.getParam("id", true);
+			this.formComponent.elements.ContactID.setValue(idFromContact);
+			this.formComponent.elements.ContactID.config.readonly = true;
+			this.formComponent.elements.ContactID.refresh();
```
Comment (origamirin) [inline sources/views/activities/formActivity.js]: Set 'this.formComponent.elements.ContactID' to variable and use it

File: sources/views/contacts/formContact.js [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+			this.image.setValues({Photo: item.Photo});
+		}
+	}
+
+	closeForm(id) {
+		this.formComponent.clear();
+		this.formComponent.clearValidation();
+		ItemDataContact.setId(null);
+		if (id) {
+			this.show(`/top/contacts?id=${id}/contacts.contactDescription`);
```
Comment (origamirin) [inline sources/views/contacts/formContact.js]: Why did you still using url for transfer id between pages? Use models like ItemDataContact

File: sources/views/contacts/contactDescription/description.js [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+		webix.confirm("Are you sure?").then(() => {
+			const idFromUrl = +this.getParam("id", true);
+			dataContacts.remove(idFromUrl);
+			let itemsForDelete = dataActivities.find(activity => activity.ContactID === idFromUrl);
+			itemsForDelete.forEach((item) => {
+				dataActivities.remove(item.id);
+			});
+			const idFirstItem = dataContacts.getFirstId();
+			if (idFirstItem) {
+				this.show(`/top/contacts?id=${idFirstItem}/contacts.contactDescription`);
```
Comment (origamirin) [inline sources/views/contacts/contactDescription/description.js]: Why did you still using url for transfer id between pages? Use models like ItemDataContact

File: sources/views/contacts/formContact.js [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+		}
+		else {
+			const idFirstItem = dataContacts.getFirstId();
+			this.show(`/top/contacts?id=${idFirstItem}/contacts.contactDescription`);
```
Comment (origamirin) [inline sources/views/contacts/formContact.js]: Why did you still using url for transfer id between pages? Use models like ItemDataContact

File: sources/views/contacts/contactDescription/activitiesAndFiles.js [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+			contactFiles.add(dataFile);
+		});
+	}
+
+	urlChange() {
+		webix.promise.all([
+			dataActivities.waitData,
+			contactFiles.waitData
+		]).then(() => {
+			const idToNumber = +ItemDataContact.getId();
```
Comment (HelgaListopad) [inline sources/views/contacts/contactDescription/activitiesAndFiles.js]: Here and below (and in all similar use-cases) convert all IDs to strigns.
id of an item is a parameter defined by backend and will not always be a number in real life.
In addition to it, Webix sometimes may give you numbers as strings.

File: sources/views/contacts/contactDescription/description.js [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+	openForm() {
+		const idContact = ItemDataContact.getId();
+		ItemDataContact.setId(idContact);
+		ItemDataContact.setItem(dataContacts.getItem(idContact));
+		this.show("./contacts.formContact");
+	}
+
+	deleteContact() {
+		webix.confirm("Are you sure?").then(() => {
+			const idContact = +ItemDataContact.getId();
```
Comment (HelgaListopad) [inline sources/views/contacts/contactDescription/description.js]: Here and below (and in all similar use-cases) convert all IDs to strigns.
id of an item is a parameter defined by backend and will not always be a number in real life.
In addition to it, Webix sometimes may give you numbers as strings.

File: sources/views/contacts/formContact.js [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+				this.closeForm();
+			});
+		}
+		else {
+			webix.message("Please check fields");
+		}
+	}
+
+	deletePhoto() {
+		this.image.setValues({Photo: avatar});
```
Comment (HelgaListopad) [inline sources/views/contacts/formContact.js]: During data saving `Photo` is extracted from this template and saved to item.
As a result, `avatar` will be saved as a part of data (which is not right). 
Set empty value here and make sure the placeholder is still rendered correctly if contact info does not contain `Photo`.

File: sources/views/activities.js [hunk 9]
```diff
@@ -60,7 +62,12 @@ export default class Activities extends JetView {
 					header: ["Contact", {content: "selectFilter"}],
 					options: dataContacts,
 					fillspace: true,
-					sort: "text"
+					sort: "text",
+					on: {
+						onBeforeFilter(id) {
+							this.getFilter(id).disabled = true;
```
[the remark this answers is not in the document]
Comment (accountF) [reply sources/views/activities.js]: Forgot to remove it. I will fix it.

**Pull request merged**